### PR TITLE
[extension] the GL_QCOM_image_processing support

### PR DESCRIPTION
**Pull request by wooyoungqcom**
The corresponding GLSL spec is here: 
https://github.com/KhronosGroup/GLSL/blob/master/extensions/qcom/GLSL_QCOM_image_processing.txt

### CODE REVIEW

**Review by arcady-lunarg — CHANGES_REQUESTED** [review]
You need to add the new capabilities to Doc.cpp, otherwise this looks good.

File: Test/baseResults/spv.tpipBlockMatchSAD.frag.out [hunk 1]
```diff
@@ -0,0 +1,123 @@
+spv.tpipBlockMatchSAD.frag
+// Module Version 10000
+// Generated by (magic number): 8000b
+// Id's are bound by 72
+
+                              Capability Shader
+                              Capability Bad
```
[the remark this answers is not in the document]
Comment (wooyoungqcom) [reply Test/baseResults/spv.tpipBlockMatchSAD.frag.out:7]: Thank you so much,.
The branch updated.

[the remark this answers is not in the document]
Comment (wooyoungqcom) [reply Test/baseResults/spv.tpipBlockMatchSAD.frag.out:7]: It seems I forgot to update the .out files.

[the remark this answers is not in the document]
Comment (wooyoungqcom) [reply Test/baseResults/spv.tpipBlockMatchSAD.frag.out]: uploaded the updated .out files

**Comment by arcady-lunarg** [comment]
Seems like CI is still failing, the SPIR-V result files say "Validation failed" while the ones in your patch don't (as they shouldn't). Is there a change in spirv-tools that this depends on? If so you'll also need to update the version of those in known_good.json.

**Comment by wooyoungqcom** [comment]
Thank you so much, yes, I think spirv-val need get updated and we are planning to upstream the corresponding changes once the glslang changes are merged.  

It seems known_good.json contains the spirv-tools commit ID that the glslang testing should use. What is the process to follow ? (I am new to the process). Do we need to upstream the spirv-tools changes first before trying to merge the glslang changes ?

**Comment by arcady-lunarg** [comment]
For reference, the spirv-tools pull request is https://github.com/KhronosGroup/SPIRV-Tools/pull/5223

**Pull request closed without merging**